{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "2de2dbc5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "dhoni\n",
      "Thu May  5 15:52:25 2022\n",
      "kohli\n",
      "Thu May  5 15:52:25 2022\n",
      "dhoni\n",
      "Thu May  5 15:52:27 2022\n",
      "kohlidhoni\n",
      "Thu May  5 15:52:29 2022\n",
      "\n",
      "Thu May  5 15:52:29 2022\n",
      "dhoni\n",
      "Thu May  5 15:52:31 2022\n",
      "kohli\n",
      "Thu May  5 15:52:33 2022\n",
      "dhoni\n",
      "Thu May  5 15:52:33 2022\n",
      "kohli\n",
      "Thu May  5 15:52:37 2022\n",
      "kohli\n",
      "Thu May  5 15:52:41 2022\n"
     ]
    }
   ],
   "source": [
    "import time\n",
    "import threading\n",
    "def fun(name,delay):\n",
    "    counter=0\n",
    "    while counter <5:\n",
    "        print(name)\n",
    "        print(time.ctime())\n",
    "        time.sleep(delay)\n",
    "        counter+=1\n",
    "\n",
    "    \n",
    "t1=threading.Thread(target=fun,args=(\"dhoni\",2))\n",
    "t2=threading.Thread(target=fun,args=(\"kohli\",4))\n",
    "t1.start()\n",
    "t2.start()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "d6d5d67e",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Assignment \n",
    "#implement pdf encryption in threads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "35627542",
   "metadata": {},
   "outputs": [],
   "source": [
    "from PyPDF2 import PdfFileWriter, PdfFileReader\n",
    "from datetime import datetime\n",
    "import threading\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "48772158",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in thread Thread-22:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/pachandan/opt/anaconda3/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/Users/pachandan/opt/anaconda3/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-3-d00bde0eaf42>\", line 19, in pdfencrypt\n",
      "NameError: name 'PyPDF2' is not defined\n",
      "Exception in thread Thread-23:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/pachandan/opt/anaconda3/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/Users/pachandan/opt/anaconda3/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-3-d00bde0eaf42>\", line 19, in pdfencrypt\n",
      "NameError: name 'PyPDF2' is not defined\n",
      "Exception in thread Thread-24:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/pachandan/opt/anaconda3/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/Users/pachandan/opt/anaconda3/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-3-d00bde0eaf42>\", line 19, in pdfencrypt\n",
      "NameError: name 'PyPDF2' is not defined\n",
      "Exception in thread Thread-25:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/pachandan/opt/anaconda3/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/Users/pachandan/opt/anaconda3/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-3-d00bde0eaf42>\", line 19, in pdfencrypt\n",
      "NameError: name 'PyPDF2' is not defined\n",
      "Exception in thread Thread-26:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/pachandan/opt/anaconda3/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/Users/pachandan/opt/anaconda3/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-3-d00bde0eaf42>\", line 19, in pdfencrypt\n",
      "NameError: name 'PyPDF2' is not defined\n",
      "Exception in thread Thread-27:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/pachandan/opt/anaconda3/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/Users/pachandan/opt/anaconda3/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-3-d00bde0eaf42>\", line 19, in pdfencrypt\n",
      "NameError: name 'PyPDF2' is not defined\n",
      "Exception in thread Thread-28:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/pachandan/opt/anaconda3/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/Users/pachandan/opt/anaconda3/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-3-d00bde0eaf42>\", line 19, in pdfencrypt\n",
      "NameError: name 'PyPDF2' is not defined\n",
      "Exception in thread Thread-29:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/pachandan/opt/anaconda3/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/Users/pachandan/opt/anaconda3/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-3-d00bde0eaf42>\", line 19, in pdfencrypt\n",
      "NameError: name 'PyPDF2' is not defined\n",
      "Exception in thread Thread-30:\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/pachandan/opt/anaconda3/lib/python3.8/threading.py\", line 932, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/Users/pachandan/opt/anaconda3/lib/python3.8/threading.py\", line 870, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-3-d00bde0eaf42>\", line 19, in pdfencrypt\n",
      "NameError: name 'PyPDF2' is not defined\n"
     ]
    }
   ],
   "source": [
    "\n",
    "\n",
    "\n",
    "namefile=\"\"\n",
    "num_of_pages=0\n",
    "\n",
    "\n",
    "#function to check whether the pdf is encrypted or not  \n",
    "import os\n",
    "files=os.listdir(\"/Users/pachandan/1_Futurense_dailywork/Day1/pdffiles\")\n",
    "\n",
    "def pdfencrypt():\n",
    "    #print(i)\n",
    "    pdfFileObj = open('/Users/pachandan/1_Futurense_dailywork/Day1/pdffiles/July018.pdf', 'rb')\n",
    "    out=PdfFileWriter()\n",
    "    pdfReader = PyPDF2.PdfFileReader(pdfFileObj)\n",
    "    num = pdfReader.numPages\n",
    "    print(num)\n",
    "\n",
    "for i in files:\n",
    "    t1=threading.Thread(target=pdfencrypt)\n",
    "    t1.start()\n",
    "    t1.join() \n",
    "  \n",
    "\n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ebb15d6c",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
